### PR TITLE
we can't pickle the new RC that uses RS secrets, so only create them …

### PR DESCRIPTION
**Pull request by dsschult**
…once inside the multiprocess pool

### CODE REVIEW

**Review by ric-evans — APPROVED** [review]
Good naming pattern with `*_creator`

File: indexer/index.py [hunk 1]
```diff
@@ -183,6 +184,10 @@ def _index(
     if not paths:
         return []
 
+    # Make FC Rest Client and IceProd Manager
+    fc_rc = fc_rc_creator()
+    manager = manager_creator()
```
Comment (blinkdog) [inline indexer/index.py:189]: I'm sure there's a Java Enterprisey BeanStrategyFactoryFactoryImpl joke in here somewhere :smile:

**Review by blinkdog — APPROVED** [review]
"All problems in computer science can be solved by another level of indirection." - David Wheeler

LGTM.

**Comment by blinkdog** [comment]
Just tried it out on `jade03`, seems to be working great. No need for workarounds in retro-disk-copy itself.
Thanks @dsschult

**Pull request merged**